### PR TITLE
Resetting activity ID to original values when set

### LINKED ISSUE #6234: SystemLogger sometimes has incorrect ActivityId
We seem to have instances where the ActivityId in Kusto is incorrect (it should be empty; but it has one, which is confusing). We need to figure out where these are coming from and how to prevent this.

@brettsam has details on queries that can show this.

### PR DESCRIPTION
**Pull request by fabiocav**
### Issue describing the changes in this PR

resolves #6234

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Comment by fabiocav** [comment]
@brettsam / @gzuber / @paulbatum  this is the tactical change I mentioned. Wanted to get his out for early feedback, but will have some tests

File: src/WebJobs.Script.WebHost/Diagnostics/FunctionsSystemLogsEventSource.cs [hunk 1]
```diff
@@ -70,12 +70,29 @@ public void LogFunctionMetricEvent(string SubscriptionId, string AppName, string
...
+            Guid.TryParse(activityId, out Guid guid);
+
+            SetCurrentThreadActivityId(guid, out Guid oldId);
+
+            return new ActivityIdScope(oldId);
+        }
+
+        public struct ActivityIdScope : IDisposable
+        {
+            public static readonly ActivityIdScope Empty = default;
```
Comment (brettsam) [inline src/WebJobs.Script.WebHost/Diagnostics/FunctionsSystemLogsEventSource.cs:84]: Is this used anywhere?

**Review by brettsam — APPROVED** [review]
Just one comment; otherwise looks good.

File: src/WebJobs.Script.WebHost/Diagnostics/FunctionsSystemLogsEventSource.cs (hunk 1 above)
Comment (fabiocav) [reply src/WebJobs.Script.WebHost/Diagnostics/FunctionsSystemLogsEventSource.cs:84]: No. It was at first. Ended up keeping it, but I can remove

**Pull request merged**